opal/corelib/array.rb

```diff
@@ -1397,7 +1399,30 @@ def inspect
   end
 
   def intersection(*arrays)
-    arrays.reduce(to_a.dup) { |a, b| a & b }
+    %x{
+      if (arrays.length === 0) {
+        return #{to_a.dup};
+      }
+      arrays = arrays.map(convertToArray);
+      if (self.length === 0) {
+        return [];
+      }
+    }
+
+    arrays = arrays.sort_by(&:length)
+    # When self is the smallest among the arrays
+    if `self.length < arrays[0].length`
+      return arrays.reduce(self, &:&)
+    end
+
+    # First, calculate intersection of argument arrays.
+    # Array#& is faster when the argument size is small.
+    # So `largest & shotest & second_shortest & ...` would be the fastest.
```

Comment:
typo: `shotest` → `shortest` (missing r)

Reply:
Fixed. Thank you!